src/webhook/mutation/pod_mutator/register.go

```diff
@@ -22,24 +22,35 @@ func registerInjectEndpoint(mgr manager.Manager, webhookNamespace string, webhoo
 	// not be ready at this point, and queries for Kubernetes objects may fail. mgr.GetAPIReader() doesn't depend on the
 	// cache and is safe to use.
 
+	eventRecorder := newPodMutatorEventRecorder(mgr.GetEventRecorderFor("dynatrace-webhook"))
 	kubeConfig := mgr.GetConfig()
+	kubeClient := mgr.GetClient()
+	apiReader := mgr.GetAPIReader()
+
+	var webhookPod corev1.Pod
+	if err := apiReader.Get(context.TODO(), client.ObjectKey{
+		Name:      webhookPodName,
+		Namespace: webhookNamespace,
+	}, &webhookPod); err != nil {
+		return errors.WithStack(err)
+	}
+
 	apmExists, err := kubeobjects.CheckIfOneAgentAPMExists(kubeConfig)
 	if err != nil {
 		return errors.WithStack(err)
 	}
 	if apmExists {
-		log.Info("OneAgentAPM object detected - DynaKube webhook won't inject until the OneAgent Operator has been uninstalled")
+		eventRecorder.sendOneAgentAPMWarningEvent(&webhookPod)
+		return errors.New("OneAgentAPM object detected - the dynatrace-webhook won't inject until the deprecated OneAgent Operator has been fully uninstalled")
```

Comment:
just a nitpick because I personally don't like contractions in error messages
```suggestion
		return errors.New("OneAgentAPM object detected - the Dynatrace webhook will not inject until the deprecated OneAgent Operator has been fully uninstalled")
```